packages/headless/src/controllers/commerce/core/pagination/headless-core-commerce-pagination.ts

```diff
@@ -116,22 +127,27 @@ export function buildCorePagination(
     },
 
     selectPage(page: number) {
-      dispatch(selectPage(page));
+      dispatch(
+        selectPage({
+          slotId,
+          page,
+        })
+      );
       dispatch(props.fetchResultsActionCreator());
     },
 
     nextPage() {
-      dispatch(nextPage());
+      dispatch(nextPage({slotId}));
```

Comment:
Should we have this action and `previousPage` accept the `slotId` directly instead?
```suggestion
      dispatch(nextPage(slotId));
```

Reply:
It would make it simpler.

On the other hand, if we ever need to add more options in the payload of those actions in the future it's going to be somewhat less elegant.

Your call, I don't feel too strongly either way 😄